tools/generate_source.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+import sys, pathlib, argparse
+import yaml
```

Comment:
We will need to make sure that standard python installations include the `pyyaml` package.  I'm not sure if that's going to be universally true.

Reply:
If we decide to generate the constants file offline, and then snapshot into e3sm, do you think it's too much to ask that the user pip installs pyyaml locally? I could prob add an inline comment reminding the user what the package name is (since it's pyyaml instead of yaml)...

Reply:
Heaven forbid our users are using `pip` instead of `conda`!  But, yes, as long as we're running this offline rather than in the build, it should be expected that users can install `pyyaml` one way or another.